source/clear-linux/tutorials/azure.rst

```diff
@@ -1,125 +1,481 @@
-create an instance of version ``12920`` of the ``basic`` offering for Clear
-Linux OS, this script creates an instance with the public hostname of
-``rob-12920-basic-1234``.
-
-::
-
-  #!/bin/bash
-  clrversion=XXXXX
-  azuser='azure_username'
-  location="westus"
-  resourceGroup="${azuser}-clr-vms"
-  offering=basic
-  # offering=containers
-  # offering=machine-learning
-  uniquesuffix=$((1 + RANDOM % 1000)) # More like "probably unique suffix"...
-
-  azure vm create --vm-size Standard_D1_v2 \
-          --admin-username ${azuser} \
-          --ssh-publickey-file ~/.ssh/id_rsa.pub \
-          --name=${azuser}-${clrversion}-${offering}-${uniquesuffix} \
-          --location=${location} \
-          --resource-group ${resourceGroup} \
-          --os-type Linux \
-          --image-urn clear-linux-project:clear-linux-os:${offering}:${clrversion}.0.0 \
-          --nic-name clr-nic1 \
-          --vnet-name ${clrversion}-${offering}-vnet \
-          --vnet-address-prefix 10.0.0.0/8 \
-          --vnet-subnet-name ${clrversion}-vnet-subnet \
-          --vnet-subnet-address-prefix 10.0.0.0/8 \
-          --plan-name ${offering} \
-          --plan-publisher clear-linux-project \
-          --plan-product clear-linux-os \
-          --public-ip-domain-name ${azuser}-${clrversion}-${offering}-${uniquesuffix} \
-          --public-ip-allocation-method "Dynamic" \
-          --public-ip-name "${clrversion}-${offering}-${uniquesuffix}"
-
-
-
-SSH Sessions
-============
-
-To keep SSH sessions to Clear Linux Guests in Azure from being dropped
-after periods of inactivity, you can give the following option to SSH via
-the command line::
-
-   -o ServerAliveInterval=180
-
-Alternatively, you can add this setting to your SSH config file as shown
-below::
-
-   Host *:
-      ServerAliveInterval 180
-
-.. [1]
-   Software and workloads used in performance tests may have been optimized
-   for performance only on Intel microprocessors. Performance tests are
-   measured using specific computer systems, components, software, operations
-   and functions. Any change to any of those factors may cause the results to
-   vary. You should consult other information and performance tests to assist
-   you in fully evaluating your contemplated purchases, including the
-   performance of that product when combined with other products. For more
-   complete information, visit http://www.intel.com/performance/datacenter.
-   Configuration: Clear Linux OS release 11130 on SKU Standard_DS3_v2 in
-   Microsoft\* Azure\*.
+#. Create a |CL| virtual machine.
+
+#. Log into the |CL| VM and use it.
+
+#. Stop the |CL| VM.
+
+#. Deallocate resources to stop incurring charges.
+
+Install MS Azure CLI 2.0 on Clear Linux
+***************************************
+
+Prerequisites for using the MS Azure CLI 2.0 on your |CL| system require you
+to have Python 2.7 or later, libffi, and OpenSSL 1.0.2 installed. If you need
```

Comment:
> +Prerequisites for using the MS Azure CLI 2.0 on your |CL| system require you
>  +to have Python 2.7 or later, libffi, and OpenSSL 1.0.2 installed. If you need

Could we recommend a way to check if these are installed or not?

Reply:
typing `python --version` will show the current version installed, for openssl, typing `openssl version` will show the version of openssl available. If you can think of an easy way to find the libffi file, let me know other than doing an `ls /usr/lib64/libffi*`